### PR TITLE
feat: HSN Code wise Item Tax

### PR DESCRIPTION
**Pull request by Anurag810**
**Overview:**
1. Added Item Tax in HSN Code master.
2. In Item master, on the selection of HSN Code, fetch item taxes as well from HSN Code master. 
3. Add a button on HSN Code to update Item Tax in all related items based on the latest item tax mentioned in HSN Code.

### CODE REVIEW

File: erpnext/regional/doctype/gst_hsn_code/gst_hsn_code.py [hunk 1]
```diff
@@ -8,3 +8,20 @@
...
+
+	taxes = frappe.parse_json(taxes)
+
+	for item in items:
+		item_to_be_updated=frappe.get_doc("Item", item.name)
+		item_to_be_updated.taxes = []
+		for tax in taxes:
+			tax = frappe._dict(tax)
+			item_to_be_updated.append("taxes", {'item_tax_template': tax.item_tax_template, 'tax_category': tax.tax_category})
+			item_to_be_updated.save()
```
Comment (rmehta) [inline erpnext/regional/doctype/gst_hsn_code/gst_hsn_code.py]: save should be outside `for`

**Review by rmehta — CHANGES_REQUESTED** [review]

File: erpnext/regional/doctype/gst_hsn_code/gst_hsn_code.js [hunk 2]
```diff
@@ -3,6 +3,26 @@
 
 frappe.ui.form.on('GST HSN Code', {
 	refresh: function(frm) {
-
+		if(! frm.doc.__islocal && frm.doc.taxes.length){
+			frm.add_custom_button(__('Update Taxes for Items'), function(){
+				frappe.confirm(
+					'Are you sure? It will overwrite taxes for all items.',
```
Comment (nabinhait) [inline erpnext/regional/doctype/gst_hsn_code/gst_hsn_code.js]: It will overwrite taxes for all items with HSN Code {0}

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/regional/doctype/gst_hsn_code/gst_hsn_code.js (hunk 2 above)
Comment (Anurag810) [reply erpnext/regional/doctype/gst_hsn_code/gst_hsn_code.js]: Done

File: erpnext/regional/doctype/gst_hsn_code/gst_hsn_code.py (hunk 1 above)
Comment (Anurag810) [reply erpnext/regional/doctype/gst_hsn_code/gst_hsn_code.py]: enqueued it for updating taxes

**Review by nabinhait — APPROVED** [review]

**Review by nabinhait — COMMENTED** [review]
@Anurag810 Please update docs

**Pull request merged**